src/classes/settings.py

```diff
@@ -111,3 +125,75 @@ def save(self):
+            return {'type':"locationImportType", 'path':"locationImportPath"}
+        if action == self.actions.EXPORT:
+            return {'type': "locationExportType", 'path': "locationExportPath"}
+        if action in [self.actions.SAVE, self.actions.LOAD]:
+            return {'type': "locationProjectType", 'path': "locationProjectPath"}
+        log.info("Action not recognized")
+        log.error("Given action is not valid %s" % action)
+        return None
+
+    def setDefaultPath(self, action: pathType, recent_path: str):
+        """ Change the path setting corresponding to the given action """
+        if not os.path.exists(recent_path):
+            return
+        if os.path.isfile(recent_path):
+            recent_path=os.path.dirname(recent_path)
+        try:
+            setting = self.pathSettings(action)
+            log.debug(f"Setting %s to %s" % (action.value, recent_path))
+            self.set(setting["path"], recent_path)
+        except:
+            log.error(f"Error setting the path of %s" % setting["path"])
+
+    def getDefaultPath(self, action: actions):
+        """
+        Returns the starting path for file browsing
+        - validates paths before returning them
+        - Actions: import, export, project open/save-as
+        - Types
+            RECENT: return the last directory where this action occurred
+            PROJECT: return the directory of the current project
+        """
+        _ = self.app._tr
+
+        default_path = ""
+        # If there isn't a current project, use the most recent one
+        project_path = self.app.project.current_filepath or ""
+        if not project_path and len(self.get("recent_projects")):
+            project_path = self.get("recent_projects")[0]
+        setting = self.pathSettings(action)
+
+        try:
+            recentOrProject = self.get(setting["type"])
+        except:
+            log.error(f"getting setting %s" % setting["type"])
+
+        if recentOrProject == self.pathType.PROJECT.value:
+            log.debug("Using Project Path")
+            default_path = project_path
+        else:
+            try:
+                log.debug("Using Recent Path")
+                default_path = self.get(setting["path"])
+            except:
+                log.error("getting setting %s" % setting["path"])
+
+        if os.path.isfile(default_path):
+            # Make sure default_path is a directory
+            default_path = os.path.dirname(default_path)
+
+        if (not default_path) or (not os.path.exists(default_path)):
+            log.debug("Default path invalid. Falling back to home directory")
+            return os.path.expanduser("~")
+
+        if action == self.actions.SAVE:
+            log.info("Adding file name for project path")
+            # Default name unless a project is currently active.
+            project_name = os.path.basename(self.app.project.current_filepath or "") or "%s.osp" % _("Untitled Project")
```

Comment:
This looks sus:
`project_name = os.path.basename(self.app.project.current_filepath or "") or "%s.osp" % _("Untitled Project")`

What does `basename("")` return? Are you trying to prevent `basename(None)`?

Reply:
Also, I'm not sure I understand adding the `project_name` to this function. It seems like the caller would add on the project_name or default name to the path returned.